generator/generator_tests/altitude_test.cpp

```diff
@@ -0,0 +1,232 @@
+
+#include "routing/road_graph.hpp"
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/altitude_loader.hpp"
+#include "indexer/classificator_loader.hpp"
+#include "indexer/index.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "geometry/point2d.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/platform.hpp"
+
+#include "platform/platform_tests_support/writable_dir_changer.hpp"
+
+#include "base/logging.hpp"
+#include "base/scope_guard.hpp"
+
+#include "std/string.hpp"
+
+using namespace feature;
+using namespace generator;
+using namespace platform;
+using namespace routing;
+using namespace tests_support;
+
+namespace
+{
+// These tests generate mwms with altitude sections and then check if altitudes
+// in the mwms are correct. The mwms are inited with different sets of features defined below.
+// There are several restrictions for these features:
+// * all of them are linear
+// * all coords of these features should be integer (it's necessary for easy implementation of MockAltitudeGetter)
+// * if accoding to one feature a point has some altitude, the point of another feature
+//   with the same coords has to have the same altitude
+// @TODO(bykoianko) Add ability to add to the tests not road features without altitude information.
+
+// Directory name for creating test mwm and temporary files.
+string const kTestDir = "altitude_generation_test";
+// Temporary mwm name for testing.
+string const kTestMwm = "test";
+
+struct Rounded3DPoint
+{
+  Rounded3DPoint(int32_t x, int32_t y, TAltitude a) : point2D(x, y), altitude(a) {}
+
+  m2::PointI point2D;
+  TAltitude altitude;
+};
+
+using TRounded3DGeom = vector<Rounded3DPoint>;
+
+TRounded3DGeom const kRoad1 = {{0, -1, -1}, {0, 0, 0}, {0, 1, 1}};
+TRounded3DGeom const kRoad2 = {{0, 1, 1}, {5, 1, 1}, {10, 1, 1}};
+TRounded3DGeom const kRoad3 = {{10, 1, 1}, {15, 6, 100}, {20, 11, 110}};
+TRounded3DGeom const kRoad4 = {{-10, 1, -1}, {-20, 6, -100}, {-20, -11, -110}};
+
+class MockAltitudeGetter : public IAltitudeGetter
+{
+public:
+  using TMockAltitudes = map<m2::PointI, TAltitude>;
+
+  MockAltitudeGetter(TMockAltitudes && altitudes) : m_altitudes(altitudes) {}
+
+  // IAltitudeGetter overrides:
+  TAltitude GetAltitude(m2::PointD const & p) override
+  {
+    m2::PointI const rounded(static_cast<int32_t>(round(p.x)), static_cast<int32_t>(round(p.y)));
+    auto const it = m_altitudes.find(rounded);
+    if (it == m_altitudes.end())
+      return kInvalidAltitude;
+
+    return it->second;
+  }
+
+private:
+  TMockAltitudes const & m_altitudes;
+};
+
+template<class T>
+vector<T> ConvertTo2DGeom(TRounded3DGeom const & geom3D)
+{
+  vector<T> dv;
+  dv.clear();
+  for (Rounded3DPoint const & p : geom3D)
+    dv.push_back(T(p.point2D));
+  return dv;
+}
+
+TAltitudes ConvertToAltitudes(TRounded3DGeom const & geom3D)
+{
+  TAltitudes altitudes;
+  altitudes.clear();
```

Comment:
Same here - no need to clear empty vector. BTW, why not to make all fields of TRounded3DGeom public and just let all clients to use them in ways they need? I.e. to collect altitudes, to convert points to something etc?